WordPress/Classes/ViewRelated/Blog/TimeZoneSelectorViewController.swift

```diff
@@ -0,0 +1,112 @@
+import UIKit
+
+class TimeZoneSelectorViewController: UITableViewController, ImmuTablePresenter {
+
+    /// users current timezone passed by SiteSettingsVC, if empty means manual offset is to be used
+    @objc var usersCurrentTimeZone: String?
+    /// users manual offset passed by SiteSettingsVC
+    @objc var usersManualOffset: NSNumber?
+
+    /// timezoneString will be set to a non empty string
+    /// if user selects anything other than Manual Offset section
+    /// manualOffset will be set if users selects anything from Manual Offset section
+    @objc var onChange: ((_ timezoneString: String, _ manualOffset: NSNumber?) -> Void)?
```

Comment:
Should `timeZoneString` be optional here?

Reply:
I'm not sure, i don't think we need to make this optional.
Since in SiteSettingsVC, there is already an if check for manualOffset being null.
If manualOffset is null, then timeZoneString has to be used, otherwise manualOffset is used

But yes i do understand, how it makes sense to either have (nil, gmtOffset) or (timeZoneString, nil) when calling this callback.
Is there some way to enforce this, that the callback has to be provided with atleast one non nil value?

Reply:
I guess you could have an enum with associated values for this

Reply:
I was hoping this would work, but then i realised the `onChange` callback in `SiteSettingsVC` is in Objective-C and an associated value enum would not be bridged to obj-c.

Reply:
Inspired by @elibud's work on PR #8347 by adding [a swift extension to SiteSettingsVC](https://github.com/wordpress-mobile/WordPress-iOS/pull/8347/files#diff-8135186a5f89ef4d57c44847559dd3c5R3)
I've also moved `showTimeZoneSelectorVC` to an extension to SiteSettingsVC in Swift, so that we can use enums with associated values
Therefore, i've refactored the code as well to use this enum wherever possible